src/docfx/build/redirection/RedirectionMap.cs

```diff
@@ -54,12 +56,11 @@ public static RedirectionMap Create(ErrorLog errorLog, Docset docset, Func<strin
 
             var redirectionsBySourcePath = redirections.ToDictionary(file => file.FilePath.Path, PathUtility.PathComparer);
 
+            CheckInvalidRedrectUrl();
             return new RedirectionMap(redirectionsBySourcePath, redirectionsByRedirectionUrl);
 
             void AddRedirections(Dictionary<string, SourceInfo<string>> items, bool redirectDocumentId = false)
```

Comment:
nit: consider turn this into a member function, this method is getting big enough